conformance/lang/expressions/unary-expression/unary-complement-expression.balt

```diff
@@ -0,0 +1,483 @@
+
+public function main() {
+    io:println(~int:MAX_VALUE);  // @output -9223372036854775808
+    io:println(~int:MIN_VALUE);  // @output 9223372036854775807
+}
+
+Test-Case: output
+Description: Test unary complement expressions with byte operand.
+Labels: byte, unary-complement
+
+public function main() {
+    byte a1 = 8;
+    int res = ~a1;
+    io:println(res); // @output -9
+
+    byte a2 = 255;
+    io:println(~a2); // @output -256
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Unsigned8 operand.
+Labels: int:Unsigned8, int:UNSIGNED8_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Unsigned8 unsignedInt = 86;
+    int res = ~unsignedInt;
+    io:println(res); // @output -87
+
+    io:println(~int:UNSIGNED8_MAX_VALUE);  // @output -256
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Unsigned16 operand.
+Labels: int:Unsigned16, int:UNSIGNED16_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Unsigned16 unsignedInt = 40123;
+    int res = ~unsignedInt;
+    io:println(res); // @output -40124
+
+    io:println(~int:UNSIGNED16_MAX_VALUE);  // @output -65536
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed16 operand.
+Labels: int:Signed16, int:SIGNED16_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed16 signedInt = 12345;
+    int res = ~signedInt;
+    io:println(res); // @output -12346
+
+    io:println(~int:SIGNED16_MAX_VALUE); // @output -32768
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Unsigned32 operand.
+Labels: int:Unsigned32, int:UNSIGNED32_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Unsigned32 unsignedInt = 12034;
+    int res = ~unsignedInt;
+    io:println(res); // @output -12035
+
+    io:println(~int:UNSIGNED32_MAX_VALUE);  // @output -4294967296
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed8 operand.
+Labels: int:Signed8, int:SIGNED8_MIN_VALUE, int:SIGNED8_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed8 signedInt = 100;
+    int res = ~signedInt;
+    io:println(res); // @output -101
+
+    io:println(~int:SIGNED8_MIN_VALUE);  // @output 127
+    io:println(~int:SIGNED8_MAX_VALUE);  // @output -128
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed16 operand.
+Labels: int:Signed16, int:SIGNED16_MIN_VALUE, int:SIGNED16_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed16 a3 = 12345;
+    int res = ~a3;
+    io:println(res); // @output -12346
+
+    io:println(~int:SIGNED16_MIN_VALUE);  // @output 32767
+    io:println(~int:SIGNED16_MAX_VALUE);  // @output -32768
+}
+
+Test-Case: output
+Description: Test unary complement expressions with Signed32 operand.
+Labels: int:Signed32, int:SIGNED32_MIN_VALUE, int:SIGNED32_MAX_VALUE, unary-complement
+
+public function main() {
+    int:Signed32 a3 = 1000;
```

Comment:
Shall we add similar test case for user defined sub types of int also

Reply:
Added in 1a98eb5fcbb2ae216fc85f8a4692fd098e9a45a9